Makefile

```diff
@@ -96,6 +96,18 @@ OPT += -momit-leaf-frame-pointer
 endif
 endif
 
+ifeq (,$(shell $(CXX) -fsyntax-only -maltivec -xc /dev/null 2>&1))
+CXXFLAGS += -DHAS_ALTIVEC
+CFLAGS += -DHAS_ALTIVEC
+HAS_ALTIVEC=1
+endif
+
+ifeq (,$(shell $(CXX) -fsyntax-only -mcpu=power8 -xc /dev/null 2>&1))
```

Comment:
This path is deliberately tied to current architecture, yes.  We needed to disallow this code from running on POWER7 and earlier models because those older hardware models don't have the special vector instructions that are used here to compute the CRC in crc32c_ppc_asm.S.  Those special vector instructions were introduced in POWER8.  

So this happens both in the Makefile as we check the compiler capabilities here, and also the architecture probe code (arch_ppc_probe function) checks the hardware capability.  

So that is why we couldn't use ifdef(__powerpc64__) in this case.